src/serviceDesks/incontact/inContactServiceDesk.ts

```diff
@@ -103,20 +103,6 @@ class InContactServiceDesk implements ServiceDesk {
       navigator.sendBeacon(`${this.SERVER_BASE_URL}/incontact/end`, JSON.stringify(this.session)); // https://golb.hplar.ch/2018/09/beacon-api.html
     });
 
-    try {
-      const request = await fetch(`${this.SERVER_BASE_URL}/incontact/queue`, {
-        method: 'POST',
-        headers: { Accept: 'application/json', 'Content-Type': 'application/json' },
-        body: JSON.stringify({ ...this.session }),
-      });
-      const output = await request.json();
-      if (output && output.queue) {
-        this.callback.updateAgentAvailability({ position_in_queue: output.queue });
-      }
-    } catch (error) {
-      return Promise.reject();
```

Comment:
It's bad practice to catch and eat errors. It can make debugging very hard if the info you need is just gone. You should always at least log them if you can. Here I would either console.error it or pass the error to reject. And just FYI it's good to avoid converting errors to just plain strings (like the error message) because errors usually come with stack traces that can also be useful.

And actually perhaps even better is to just remove the try...catch entirely. If you do that, this code will simply throw the error which will reject the async Promise anyway. Although I think there's perhaps there's something not good here because if this code throws an error, it'll be caught by the outer try...catch which will *not* happen if you return Promise.reject. So if you remove the try...catch, the code might work differently than it does now. That Promise.reject will bypass that outer try...catch. I think it's not a good idea to mix Promise.reject and try...catch in the same code because you could easily get unexpected behaviors.

Another consequence of the `return Promise.reject` is that `this.poller.stop` is not set to true. I think given the current code that doesn't actually matter since the poller stops anyway but another example and a potential landmine someone might step on with changes in the future. Also the `setErrorStatus` code won't be called - that seems like it could be wrong but maybe not.

So in general, you should probably review the error handling in here :-).

Reply:
basically i ended up changing the rejects to `throw new Error(error)` and passed in the error. For specifically the queuing section, i left the try catch block and let potential errors slide by, since the agent availability call shouldn't necessarily need to disrupt the entire polling mechanism and shut down the chat. I just put a `console.error()` for that error handling.